gh-114616: Improve docs regarding changes to caches representation in dis (#120033)
python/cpython: Doc/library/dis.rst

      Added the *show_caches* and *adaptive* parameters.

   .. versionchanged:: 3.13
      The *show_caches* parameter is deprecated and has no effect. The iterator
      generates the :class:`Instruction` instances with the *cache_info*

>       field populated (regardless of the value of *show_caches*) and it does not
>       generate separate items for the cache entries as it did before.

Author: ```suggestion
      field populated (regardless of the value of *show_caches*) and it no longer
      generates separate items for the cache entries.
```
